### PR TITLE
Added example for Text Generation with GPT2

### PR DESCRIPTION
**Pull request by AdarshKumar712**
I have created a simple example on Text generation with HuggingFace gpt2 model. Please have a look and suggest if any changes required.

@chengchingwen

### CODE REVIEW

File: example/GPT2_TextGeneration/text_generation.jl [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+using Transformers
+using Transformers.Basic
+using Transformers.GenerativePreTrain
+using Transformers.Pretrain
+using Transformers.HuggingFace
+
+isdir("model_vocab")||mkdir("model_vocab")
+
+# This is a temporary fix, will be updated once Tokenizer API is ready
+HTTP.download("https://s3.amazonaws.com/models.huggingface.co/bert/gpt2-merges.txt", "./model_vocab/bpe.out")
```
Comment (chengchingwen) [inline example/GPT2_TextGeneration/text_generation.jl:12]:
```suggestion
isfile("model_vocab/bpe.out") || HTTP.download("https://s3.amazonaws.com/models.huggingface.co/bert/gpt2-merges.txt", "./model_vocab/bpe.out")
```

File: example/GPT2_TextGeneration/text_generation.jl [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+HTTP.download("https://s3.amazonaws.com/models.huggingface.co/bert/gpt2-vocab.json", "./model_vocab/vocab.json")
```
Comment (chengchingwen) [inline example/GPT2_TextGeneration/text_generation.jl:13]:
```suggestion
isfile("model_vocab/vocab.json") || HTTP.download("https://s3.amazonaws.com/models.huggingface.co/bert/gpt2-vocab.json", "./model_vocab/vocab.json")
```

File: example/GPT2_TextGeneration/text_generation.jl [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    push!(text_list, decoder[tokens[i]])
+  end
+  text = replace(join(text_list), "Ġ"=> " ")
+  return text
+end
+
+text_token_ids = generate_text(context = "Fruits are very good for "; max_length = 100)
+gen_text = decode_text(text_token_ids)
+print("\n\nGenerated Text: ")
+print(gen_text)
```
Comment (chengchingwen) [inline example/GPT2_TextGeneration/text_generation.jl:74]: Better wrap the generation into one function so people can easily test with.

**Comment by AdarshKumar712** [comment]
Done!

File: example/GPT2_TextGeneration/text_generation.jl [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+end
+
+function generate()
+  text_token_ids = generate_text(context = "Fruits are very good for "; max_length=100)
+  gen_text = decode_text(text_token_ids)
+  print("\n\nGenerated Text: ")
+  print(gen_text)
+end
+
+generate()
```
Comment (chengchingwen) [inline example/GPT2_TextGeneration/text_generation.jl:78]:
```suggestion
function generate(prompt)
  text_token_ids = generate_text(context = prompt; max_length=100)
  gen_text = decode_text(text_token_ids)
  print("\n\nGenerated Text: ")
  println(gen_text)
end

generate( "Fruits are very good for ")

```

Comment (AdarshKumar712) [reply example/GPT2_TextGeneration/text_generation.jl:78]: I think then `max_length` also be given as parameter to generate function.

**Pull request merged**